cmake/version.cmake

```diff
@@ -76,6 +76,31 @@ execute_process(
   RESULT_VARIABLE "VERSION_IS_SEMVER"
 )
 
+# Produce a valid version for the Python package
+set(CCF_PYTHON_VERSION "${CCF_RELEASE_VERSION}")
```

Comment:
I think I'd have been tempted to do this in Python to be easily able to add a few unit tests.

Reply:
That's a really good point. I had convinced myself that CMake was a hammer, when it is clearly made from silly string.

Reply:
This is now done by a Python script, so handles even more cases with helpful auto-canonicalisation by the Python `packaging.version`. In particular, it helpfully turns the git describe `1.2.3-42-deadbeef` (where 42 is the number of commits since `1.2.3`) into `<Version('1.2.3.post42+deadbeef')>`.